conftest.py

```diff
@@ -62,6 +62,27 @@ def skip_warnings(pytestconfig):
     return skip_warnings
 
 
+@pytest.fixture(scope="session")
+def require_warnings_skipped(skip_warnings):
```

Comment:
I think we should rename the CLI option to something like `--no-intentional-warnings` and rename the `skip_warnings` fixture to something like `skip_intentional_warnings`.

After this refactor, as they're currently named, I'm having trouble keeping straight the semantics of "intentional warnings" (warnings raised as part of the test suite, to probe `stderr`) versus "unintentional warnings" (*real* warnings from, e.g., deprecations, that we aren't raising intentionally as part of testing and don't want to be silenced, so that they can be addressed).

Reply:
I dont think "intentional" makes it any clearer.

They are all intentional - even the ones triggers in stdlib code are required parts of the test suite.

I wasnt sure how to resolve this, but I think I now have something less confusing.